deploy/runtools/runtime_config.py

```diff
@@ -225,6 +251,78 @@ def __str__(self) -> str:
         return """RuntimeHWConfig: {}\nDeployTriplet: {}\nAGFI: {}\nCustomRuntimeConf: {}""".format(self.name, self.deploytriplet, self.agfi, str(self.customruntimeconfig))
 
 
+
+
+class RuntimeBuildRecipeConfig(RuntimeHWConfig):
+    """ A pythonic version of the entires in config_build_recipes.yaml """
+
+    def __init__(self, name: str, build_recipe_dict: Dict[str, Any], default_metasim_host_sim: str) -> None:
+        self.name = name
+        self.agfi = "Metasim" # for __str__ to work
+        self.deploytriplet = build_recipe_dict['DESIGN'] + "-" + build_recipe_dict['TARGET_CONFIG'] + "-" + build_recipe_dict['PLATFORM_CONFIG']
+
+        self.customruntimeconfig = build_recipe_dict['metasim_customruntimeconfig']
+        self.customruntimeconfig = self.customruntimeconfig if self.customruntimeconfig != "None" else None
+        # note whether we've built a copy of the simulation driver for this hwconf
+        self.driver_built = False
+        self.metasim_host_simulator = default_metasim_host_sim
+
+        self.platform = "f1"
+        self.driver_name_prefix = ""
+        self.driver_name_suffix = ""
+        if self.metasim_host_simulator in ["verilator", "verilator-debug"]:
+            self.driver_name_prefix = "V"
+        if self.metasim_host_simulator in ['verilator-debug', 'vcs-debug']:
+            self.driver_name_suffix = "-debug"
+
+        self.local_driver_base_dir = LOCAL_DRIVERS_GENERATED_SRC
+
+        dramsim_pair = (self.get_local_driver_dir() + "dramsim2_ini", "")
+        self.additional_required_files = [dramsim_pair]
+
+        self.driver_build_target = self.metasim_host_simulator
+        self.driver_type_message = "Metasim"
+
+    def get_boot_simulation_command(self,
+            slotid: int,
+            all_macs: Sequence[MacAddress],
+            all_rootfses: Sequence[Optional[str]],
+            all_linklatencies: Sequence[int],
+            all_netbws: Sequence[int],
+            profile_interval: int,
+            all_bootbinaries: List[str],
+            all_shmemportnames: List[str],
+            tracerv_config: TracerVConfig,
+            autocounter_config: AutoCounterConfig,
+            hostdebug_config: HostDebugConfig,
+            synthprint_config: SynthPrintConfig,
+            extra_plusargs: str = "",
+            extra_args: str = "") -> str:
+        """ return the command used to boot the meta simulation. """
+        full_extra_plusargs = " +fesvr-step-size=128 +dramsim +max-cycles=100000000" + extra_plusargs
```

Comment:
I can see people trying to modify these plusargs (esp. max cycles). Would it make sense to move these to the YAML file so that people could modify this? Or if you specify it again, does the latter plusarg take precedence?

Reply:
This is good point. 

I think earlier plus args will take precedence but we should avoid repetition if possible.

Reply:
I think until we come up with more general +arg handling these should just be exposed in the YAML.

Reply:
fixed